binary/expression.proto

```diff
@@ -216,6 +215,101 @@ message Expression {
       repeated string prerequisite = 2;
     }
   }
+
+  message ReferenceSegment {
```

Comment:
Can you add a small piece of documentation to the toplevel messages here? I think that's `ReferenceSegment`, `MaskExpression` and `FieldReference` if I'm reading it correctly.